### PR TITLE
Add constructor functions for model and tokenizer of MonoBERT/T5

### PR DESCRIPTION
**Pull request by yuxuan-ji**
This PR is intended to try and resolve the issue raised in https://github.com/castorini/pygaggle/issues/86#issuecomment-691696129 and closes https://github.com/castorini/pygaggle/issues/86
i.e.
> Wouldn't it make sense to use this new abstraction so we don't have multiple code paths (that might subtly diverge down the road...)? And same with MonoBERT?

I'm proposing we add constructor functions  for the model and tokenizer used in MonoBERT/T5 so that the user doesn't need to manually initialize them to the right class

with this change, i'm also removing MonoBERT/T5's ability to take in strings as parameters, as it becomes redundant

thoughts on this change? I think it might be a more sane solution that what I had before

### CODE REVIEW

File: pygaggle/rerank/transformer.py [hunk 1]
```diff
@@ -27,19 +27,27 @@
...
-        self.model = model_name_or_instance
-
-        if isinstance(tokenizer_name_or_instance, str):
-            tokenizer_name_or_instance = T5BatchTokenizer(AutoTokenizer.from_pretrained(tokenizer_name_or_instance), batch_size=8)
-        self.tokenizer = tokenizer_name_or_instance
-
+                 model: T5ForConditionalGeneration = None,
+                 tokenizer: QueryDocumentBatchTokenizer = None):
+        self.model = model or self.get_model()
+        self.tokenizer = tokenizer or self.get_tokenizer()
```
Comment (ronakice) [inline pygaggle/rerank/transformer.py:33]: I'm not sure but is such code standard python practice?

Comment (yuxuan-ji) [reply pygaggle/rerank/transformer.py:33]: @ronakice, is your concern about the use of  the `or` operand?  I've seen this being used as a shorthand to
```python
self.model = model if model else self.get_model()
```
so I don't think there's any issues there. Or is it more about the PR in general?

**Review by ronakice — APPROVED** [review]
Thanks a lot for this @yuxuan-ji :) , merging!

**Comment by ronakice** [comment]
@yuxuan-ji  actually can you resolve these merge conflicts?

**Pull request merged**